cmake/scripts/common/Macros.cmake

```diff
@@ -785,7 +789,9 @@ macro(find_addon_xml_in_files)
     string(REPLACE "${CORE_SOURCE_DIR}/" "" xml_name ${xml_name})
 
     list(APPEND ADDON_XML_DEPENDS "${CORE_SOURCE_DIR}/${xml_name}/addon.xml.in")
-    list(APPEND ADDON_XML_OUTPUTS "${CMAKE_BINARY_DIR}/${xml_name}/addon.xml")
+    if(NOT ${xml_name} MATCHES ${filter})
```

Comment:
As filter could be a list MATCHES is wrong.
You should ensure that filter either is always a string or always a list.